subiquity/controllers/error.py

```diff
@@ -0,0 +1,177 @@
+# Copyright 2019 Canonical, Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import datetime
+import enum
+import json
+import logging
+import os
+
+import apport
+import apport.crashdb
+import apport.hookutils
+
+import attr
+
+from subiquitycore.controller import BaseController
+
+
+log = logging.getLogger('subiquity.controllers.error')
+
+
+class ErrorReportState(enum.Enum):
+    INCOMPLETE = _("INCOMPLETE")
+    DONE = _("DONE")
+    ERROR = _("ERROR")
+
+
+class ErrorReportKind(enum.Enum):
+    BP_FAIL = _("Block device probe failure")
+    RBP_FAIL = _("Disk probe failure")
```

Comment:
BLOCK_PROBE_FAIL
DISK_PROBE_FAIL

Would make reading the enums in the rest of the code easier.

Reply:
OK